Sources/_MatchingEngine/Regex/Printing/PrintAsCanonical.swift

```diff
@@ -34,7 +34,7 @@ extension AST.Node {
   ) -> String {
     var printer = PrettyPrinter()
     printer.printAsCanonical(
-       self,
+       .init(self, globalOptions: nil),
        delimiters: delimiters,
        terminateLine: terminateLine)
     return printer.finish()
```

Comment:
My intuition is that we then want to hoist this to `AST` instead of on nodes. The render which takes various options makes sense on the AST so it picks up global options, etc, and the recursive logic is on node.

Reply:
There is an overload for `AST` too, this overload is mainly for convenience e.g a few call sites doing `throw unsupported(node.renderAsCanonical())`. Though we could reduce the duplication here by having it call into the other overload, let me change that.